org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -34,8 +35,16 @@
 	private DotnetNewWizardPage wizardPage;
 	@Override
 	public void init(IWorkbench workbench, IStructuredSelection selection) {
-		// TODO
 		wizardPage = new DotnetNewWizardPage();
+
+		Object firstElement = selection.getFirstElement();
+		if (firstElement instanceof IResource) {
```

Comment:
> If there are a lot of working sets and projects, it may take a while to search for containing working sets.

Maybe, however I believe the operations are fast enough and the usual number of projects/working sets low enough (max I've seen was about 20 working sets and 200 projects) to consider this issue as important. I suggest you ignore the potential performance issue right now as it's not likely to happen and as the other workarounds you propose would be way harder to implement.
Let's first face the issue in action before we try to solve it.

About the proposal:
1. wouldn't improve performance. Just it moves the user "wait" on another step, but user would still have to wait
2. is probably the nicer one
3. a UI must do what the user has asked for. If we do something else, users become angry.

But again, in that case, I suggest we go for 4. We don't care about this very unlikely issue and compute the working set synchronously in UI, and if the issue is perceived one day, we'll improve it then.